src/frontend/src/binder/select.rs

```diff
@@ -435,6 +503,49 @@ impl Binder {
             }
         })
     }
+
+    /// Uses a literal value to look up the ID of an Object (e.g. a table). If the literal is an
+    /// integer (int16, int32, or int64) this will just return that number as an ObjectID value.
+    /// If the literal is a varchar, this will look in the Catalog for an object with a name that
+    /// matches the varchar and return its Object ID value; of no match is found then an error is
+    /// returned.
+    fn lookup_object_id(&mut self, arg: &Literal) -> Result<TableId> {
+        match arg
+            .get_data()
+            .as_ref()
+            .ok_or_else(|| ErrorCode::BindError("No Value".to_string()))?
+        {
+            ScalarImpl::Int16(id) => Ok(TableId::new(*id as u32)),
+            ScalarImpl::Int32(id) => Ok(TableId::new(*id as u32)),
+            ScalarImpl::Int64(id) => Ok(TableId::new(*id as u32)),
+            ScalarImpl::Utf8(name) => {
+                // We use the full parser here because this function needs to accept every legal way
+                // of identifying a table in PG SQL as a valid value for the varchar
+                // literal.  For example: 'foo', 'public.foo', '"my table"', and
+                // '"my schema".foo' must all work as values passed pg_table_size.
+                let mut tokenizer = Tokenizer::new(name);
```

Comment:
It looks a littble bit weird to me to use parser in the binder. cc @xiangjinwu Is this a good practice? Any suggestion?

Reply:
Yes it is wired, but also somewhat reasonable because we are doing introspection here. PG even have a SQL function [`parse_ident`](https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/misc.c#L696) but it looks similar to [`SplitIdentifierString`](https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/varlena.c#L3711). My suggestion would be to extract this to a utility function outside binder, and still use parser inside to avoid the burden of maintaining duplicate implementation. This does sound like cheating and makes no difference at run time. 😂

Reply:
I thought about writing something custom for this, but we need to support any valid representation of a table which would mean building a parallel parser for just table names that are passed as varchar literals and, critically, having to always remember that any change to the SQL parser would have to be reflected in the parallel parser. I couldn't come up with a good reason that would be worth that maintenance load.

Putting into a helper function makes sense to me: what would be the best place to put it?